SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -682,33 +680,22 @@ FRPCErrorInfo USpatialSender::SendRPC(const FPendingRPCParams& Params)
 	}
 
 	const FRPCInfo& RPCInfo = ClassInfoManager->GetRPCInfo(TargetObject, Function);
-	bool bUseRPCRingBuffer = GetDefault<USpatialGDKSettings>()->UseRPCRingBuffer();
 
 	if (RPCInfo.Type == ERPCType::CrossServer)
 	{
 		SendCrossServerRPC(TargetObject, Function, Params.Payload, Channel, Params.ObjectRef);
 		return FRPCErrorInfo{ TargetObject, Function, ERPCResult::Success };
 	}
 
-	if (bUseRPCRingBuffer && RPCService != nullptr)
+	checkf(RPCService != nullptr, TEXT("Attempting to send an RPC with a null RPCService"));
+	if (SendRingBufferedRPC(TargetObject, Function, Params.Payload, Channel, Params.ObjectRef))
 	{
-		if (SendRingBufferedRPC(TargetObject, Function, Params.Payload, Channel, Params.ObjectRef))
-		{
-			return FRPCErrorInfo{ TargetObject, Function, ERPCResult::Success };
-		}
-		else
-		{
-			return FRPCErrorInfo{ TargetObject, Function, ERPCResult::RPCServiceFailure };
-		}
+		return FRPCErrorInfo{ TargetObject, Function, ERPCResult::Success };
 	}
-
-	if (Channel->bCreatingNewEntity && Function->HasAnyFunctionFlags(FUNC_NetClient))
```

Comment:
still need this right?

Reply:
Only if RPCService == nullptr, is this a possible case?

Reply:
True, I guess one divergence here is that we would now return ERPCResult::RPCServiceFailure instead of attempting to send a legacy rpc